src/parse/string/shift/operand/date.ts

```diff
@@ -0,0 +1,57 @@
+import { typeNode } from "../../../../main.js"
+import { throwParseError } from "../../../../utils/errors.js"
+import type { DynamicState } from "../../reduce/dynamic.js"
+import type { state, StaticState } from "../../reduce/static.js"
+import type { Scanner } from "../scanner.js"
+import {
+    type EnclosingChar,
+    untilLookaheadIsClosing,
+    writeUnterminatedEnclosedMessage
+} from "./enclosed.js"
+
+export type parseDate<
```

Comment:
This feels like a lot of duplicated code for something that is very similar to parse enclosed.

Remember I mentioned an approach where we could have enclosed be a Record of opening to closing chars instead? Let's see if we can implement it using an approach like that